examples/robot-assistant/server/src/main.rs

```diff
@@ -0,0 +1,128 @@
+mod robot;
+
+use tide_websockets::{Message, WebSocket};
+use async_std::prelude::*;
+use async_std::task::spawn;
+use futures::future::join;
+use anyhow::Result;
+use std::io::{Error, ErrorKind};
+use tide::sessions::SessionMiddleware;
+use tide::sessions::CookieStore;
+use uuid::Uuid;
+use std::process::Command;
+use robot::Robot;
+use robot::State;
+
+#[async_std::main]
+async fn main() -> Result<()> {
+    println!("server");
+    tide::log::start();
+    let mut app = tide::new();
+    app.with(SessionMiddleware::new(
+        CookieStore::new(),
+        b"936DA01F9ABD4d9d80C702AF85C822A8",
+    ));
+
+    app.with(tide::utils::Before(
+        |mut request: tide::Request<()>| async move {
+            let session = request.session_mut();
+            let topic_id = if let Some(topic_id) = session.get("topic_id") {
+                topic_id
+            } else {
+                let my_uuid = Uuid::new_v4();
+                let topic_id = format!("{}", my_uuid.to_simple());
+                // FIXME
+                Command::new("cargo")
```

Comment:
move topic set up out of application.  assume topic is setup correctly.
Can add topic initialization in the cargo makefile

Reply:
I have a question about the topic creation. How can create a "session" topic and query logs in that topic with a sessionid. Similar concept in dynamodb's secondary index.